src/v/utils/rwlock.h

```diff
@@ -36,6 +36,8 @@ class rwlock_unit {
     rwlock_unit(const rwlock_unit&&) = delete;
     rwlock_unit() = delete;
 
+    explicit operator bool() const noexcept { return _lock != nullptr; }
```

Comment:
if we are leaving in operator bool() let's make it behave similar to `semaphore_units::operator bool` and have it check that units are also held.

Reply:
` _lock != nullptr` means exactly that; I'll make ctor private & friend to enforce it